GOVERNANCE.md

```diff
@@ -42,6 +45,33 @@ Maintainers have right to nominate contributors to become maintainers and vote o
 
 Individuals can be maintainers to multiple Eiffel Community Open Source Projects.
 
+
+## Community Roles
+
+Eiffel Community currently has two roles relevant to community governance and support.
+
+### Election Officers
+
+Eiffel Community Election Officers are the community members who are appointed by Eiffel Technical Committee to conduct community elections.
+
+The term for Election Officers is **one year**.
+
+Responsibilities of Election Officers are
+
+* Determine the list of community members who are eligible to vote and run in community elections.
```

Comment:
What are 'community elections'? Is it only elections to the Eiffel TC or also something else? In the Maintainers section above it is stated that  Maintainers are eligible to vote on TC elections. Can the Election Officers also appoint other community members to vote on TC elections? Could we make it simple and just state that all Maintainers can vote in all community elections?
What does it then mean to 'run' in a community election? To be elected to TC? If so, again, any Maintainer could do that, right?

Reply:
> What are 'community elections'? Is it only elections to the Eiffel TC or also something else? 

It is only Eiffel TC elections - at least for the timebeing. Will explicitly state it.

> In the Maintainers section above it is stated that Maintainers are eligible to vote on TC elections. 

All the contributors, regardless if they are maintainers or not, are eligible to vote on elections.

   _Anyone who is a contributor to one or more projects hosted by Eiffel Community is eligible to vote on Technical Committee Elections._

Only the maintainers are eligible to nominate or be nominated in elections.

I couldn't find a place that says only maintainers are eligible to vote on TC elections but I might have missed.
Could you please point that place so we fix that part?

> Could we make it simple and just state that all Maintainers can vote in all community elections?

I think all contributors being eligible to vote covers this part without additional update.

> Can the Election Officers also appoint other community members to vote on TC elections?

Yes with conditions. Please see my response at the end.

> What does it then mean to 'run' in a community election? To be elected to TC? If so, again, any Maintainer could do that, right?

Run in this context means "being a nominee" in the election to be elected to TC as you highlight.

When it comes to who can "vote" and "run".
Here is some more thoughts on this/

- **Who can "vote"**: The document states all the contributors are eligible to vote but I've been thinking about this for a while since it will be a bit difficult to determine the list of contributors, thus making TC elections slightly heavier. (we need to crawl git repos/issues to see who contributed) We could perhaps make this even simpler and say "Anyone who is subscribed to Eiffel Maillist is eligible to vote." This will hopefully capture everyone who is either maintainer and contributor and also give others a reason to be more active in the community, given that their opinion in the elections are deemed valuable.

- **Who can "run"**:  In other communities, only committers/maintainers are eligible to run but not always. Some communities use metrics to determine who can run in TC elections, regardless of their maintainer status. No of PRs, issues, wiki edits, mails to maillist, IRC messages and so on are taken into account and if you have more than X cumulatively (e.g., 20), you are eligible to run. Some communities take this even further since this approach still doesn't capture everyone such as some contributors who are actually contributing to the community without being committer/maintainer who fail to meet metric criteria. (e.g., people who contribute with valuable ideas during meetings for example.) For this type of situations, some communities have exception processes in place. Other community members could file an exception for such contributors to TC, requesting certain individual(s) to be  eligible to run for a seat in TC. See an example [here](https://lists.opnfv.org/g/opnfv-tech-discuss/message/23271). We can stick to maintainer status or perhaps add a sentence stating that "Upon completion of TC election preparation phase and the list of people who are eligible to run in elections is announced on Eiffel maillist, maintainers have X days to file an exception to TC, requesting certain individuals to be made eligible to run in elections.". I think we should have this clause to ensure we don't exclude people with contributions in different ways without being maintainer from running in elections.

Reply:
> We could perhaps make this even simpler and say "Anyone who is subscribed to Eiffel Maillist is eligible to vote."
I guess that would be ok, as only maintainers can nominate or be nominated in elections. If we choose to make all subscribers able to vote, I'd say that we should also consider anyone on the maillist as part of the community, and anyone _not_ on the maillist is _not_ part of the community, or would that be discriminating to people that don't want to receive all community mails?

> maintainers have X days to file an exception to TC, requesting certain individuals to be made eligible to run in elections.
Yes, I believe there should be an option to nominate people that are not maintainers of any Eiffel repo, through an exemption request

Reply:
> I guess that would be ok, as only maintainers can nominate or be nominated in elections. If we choose to make all subscribers able to vote, I'd say that we should also consider anyone on the maillist as part of the community, and anyone _not_ on the maillist is _not_ part of the community, or would that be discriminating to people that don't want to receive all community mails?

I think we need to put a basic expectation to community members like how it is done in open source communities in general - everyone is encouraged to subscribe to community maillists.

However, what you say is important and valid as well - people have right **not to** subscribe maillists.
In that case, they will miss nomination period unless someone else forwards the mail to them or nominates them.
They will not be able to cast their votes in the elections either since they will not know it is happening.

I wouldn't call this discrimination since there isn't much we can do about it except putting an effort to extracting data from multiple systems - GitHub commit history, GitHub issues, maillist, and so on which I'm not sure how it can be done unless someone puts an effort to do that or we find an open source tool that does that or we find a hosting organization. (hosting organizations such as CDF or Open Infra Foundation provide these services to the communities they provide home_

What I can suggest here is that the TC that will be seated after the first elections work on this further to make elections as inclusive as possible. Being inclusive is critical and important but how much effort we reasonably could put in place is something that needs to be discussed.